app/assets/javascripts/discourse/widgets/hamburger-menu.js.es6

```diff
@@ -176,18 +176,26 @@ export default createWidget("hamburger-menu", {
   },
 
   listCategories() {
-    const hideUncategorized = !this.siteSettings.allow_uncategorized_topics;
-    const isStaff = Discourse.User.currentProp("staff");
+    const maxCategoriesToDisplay = 6;
+    const categoriesList = this.site.get("categoriesList");
```

Comment:
I don't think this works quite right ... you first need to get all the categories ordered by count (in some conditions) and then do the filtering down. 

This change makes the list of categories we show arbitrary afaik, especially since we suppress subcategories.